src/test/java/com/hillel/items_exchange/controller/CategoryControllerIntegrationTest.java

```diff
@@ -0,0 +1,277 @@
+package com.hillel.items_exchange.controller;
+
+import com.github.database.rider.core.api.dataset.DataSet;
+import com.github.database.rider.core.api.dataset.ExpectedDataSet;
+import com.github.database.rider.spring.api.DBRider;
+import com.hillel.items_exchange.dto.CategoryDto;
+import com.hillel.items_exchange.exception.InvalidDtoException;
+import com.hillel.items_exchange.util.CategoryControllerIntegrationTestUtil;
+import org.hibernate.boot.model.naming.IllegalIdentifierException;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.http.MediaType;
+import org.springframework.security.test.context.support.WithMockUser;
+import org.springframework.test.context.jdbc.Sql;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+
+import javax.persistence.EntityNotFoundException;
+import javax.transaction.Transactional;
+
+import static com.hillel.items_exchange.util.TestUtil.asJsonString;
+import static org.hamcrest.CoreMatchers.instanceOf;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.hasSize;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;
+import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@SpringBootTest
+@DBRider
+@AutoConfigureMockMvc
+@Sql(executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD, scripts = "classpath:index-reset.sql")
+public class CategoryControllerIntegrationTest extends CategoryControllerIntegrationTestUtil {
+
+    private final MockMvc mockMvc;
+
+    @Autowired
+    public CategoryControllerIntegrationTest(MockMvc mockMvc) {
```

Comment:
Would you tried to use @RequiredArgsConstructor here?

Reply:
No, it fails on this way((